app/components/GMap.jsx

```diff
@@ -1,81 +1,120 @@
+          icon: PropTypes.string,
+          message: PropTypes.string
+        })),
+        snapToUserLocation: PropTypes.bool
+      })
     }
   }
 
   static get defaultProps() {
     return {
-        center: {
+      config: {
+        initialCenter: {
           lat: 29.975588,
-          lng: -90.102682 },
-        message: "A Message"
+          lng: -90.102682
+        },
+        initialZoom: 10
+      }
     }
   }
 
   constructor(props){
     super(props);
     this.state = {
-      zoom: 11,
-      infoWindowIsOpen: true
+      center: null
     };
   }
 
   loadMap() {
     if (this.state.scriptLoaded) {
-      // lets map autocenter on user's location (if the user enables it)
-      // which takes a while, so the map is created and goes to the initial center first
-      if (this.props.findUserLocation && navigator.geolocation) {
-        navigator.geolocation.getCurrentPosition( (position) => {
-          this.setState({
-            center: this.mapCenter(position.coords.latitude, position.coords.longitude)
-          });
-          this.moveMap("Got it!");
-        }, () => this.infoWindow.setContent("Couldn't find your location :("))
+      if (this.props.config && this.props.config.snapToUserLocation && navigator.geolocation) {
+        this.getUserLocation()
       } else {
         this.setState({
-          center: this.mapCenter(this.props.center.lat, this.props.center.lng)
+          center: this.mapCenter(this.props.config.initialCenter.lat, this.props.config.initialCenter.lng)
         })
       }
-      // create the map, marker and infoWindow after the component has
+      // create the map and markers after the component has
       // been rendered because we need to manipulate the DOM for Google =(
-      this.map = this.createMap(this.state.center);
-      this.marker = this.newMarker(this.state.center, this.map, this.props.markerImage);
-      this.infoWindow = this.newInfoWindow(this.map, this.marker, this.props.message);
-
-      // have to define google maps event listeners here too
-      // because we can't add listeners on the map until its created
-      google.maps.event.addListener(this.map, 'zoom_changed', () => this.handleZoomChange());
-      google.maps.event.addListener(this.marker, 'click', () => this.toggleInfoWindow())
+      this.map = this.createMap(this.props.config.initialCenter);
+      if (this.props.config && this.props.config.markers) {
+        this.markers = this.createMarkers(this.props.config.markers);
+        if (this.props.config.legend) {
+          this.createLegend(this.props.config.icons);
+        }
+      }
     }
   }
 
   // clean up event listeners when component unmounts
   componentDidUnMount() {
-    google.maps.event.clearListeners(map, 'zoom_changed')
+    google.maps.event.clearListeners(map, 'click')
+  }
+
+  createLegend(icons) {
+    let legend = this.refs.legend;
+    for (let key in icons) {
+      let type = icons[key];
```

Comment:
Had to do it as `const type = icons[key], name = type.name, icon = type.image` since the latter two depend on `type` being declared already, but 👍.  

`let`s are switched to `const`s - I've read that it's bad practice to use `const` in loops, but it was probably just someone being cranky about their preferred style.